crates/stable-diffusion-bot/src/bot/handlers/image.rs

```diff
@@ -133,25 +133,58 @@ impl Response {
 struct MessageText(String);
 
 impl MessageText {
-    pub fn new_with_infotxt(prompt: &str, infotxt: &str) -> Self {
+    pub fn new_with_imginfo(prompt: &str, infotxt: &ImgInfo) -> Self {
         use teloxide::utils::markdown::escape;
+
         lazy_static! {
-            static ref RE: Regex = Regex::new(r"(?P<key>[^,]+): (?P<value>[^,]+),? ?").unwrap();
+            static ref RE: Regex = Regex::new(r"Model: (.+?),").unwrap();
         }
-        Self(format!(
-            "`{}`\n\n{}",
-            escape(prompt),
-            RE.replace_all(
-                escape(infotxt.strip_prefix(prompt).unwrap_or(infotxt)).as_str(),
-                "$key: `$value`\n",
+
+        let model = RE
+            .captures(
+                infotxt
+                    .infotexts
+                    .clone()
+                    .unwrap_or_default()
+                    .first()
+                    .unwrap()
+                    .as_str(),
             )
-            .trim()
-        ))
-    }
+            .unwrap()
+            .get(1)
+            .unwrap()
+            .as_str()
+            .to_string();
 
-    pub fn new(prompt: &str) -> Self {
-        use teloxide::utils::markdown::escape;
-        Self(format!("`{}`", escape(prompt),))
+        Self(format!(
+            "{}{}",
+            match infotxt.negative_prompt.clone() {
+                Some(n_prompt) if !n_prompt.is_empty() => {
+                    format!(
+                        "`{}`\n\nNegative prompt: `{}`\n",
+                        escape(prompt),
+                        escape(&n_prompt)
+                    )
+                }
+                _ => format!("`{}`\n\n", escape(prompt)),
+            },
+            [
+                format!("Steps: `{}`", infotxt.steps.unwrap(),),
```

Comment:
I generally try to avoid bare `.unwrap()` calls as it can lead to undesirable panic crashes. In this case, it's not a big deal if data is missing so we definitely shouldn't crash; we could just omit any missing parameters from the string.

Maybe something like:

```rust
infotxt.steps.map(|s| format!("Steps: `{s}`"));
```
(repeat for each param)

Then you'll have a slice of `Option`s, which you can `.flatten()` to remove the `None`s.

Reply:
Removed all unwraps, now it shoud be safer

Reply:
Thanks!